### PR TITLE
Introduce Audit Tracker (Track Creator Updater)

### PR DESCRIPTION
**Pull request by xjunior**
- Initialize data tracker gem
- Add dummy app for specs
- Ignore Gemfile.lock
- Install rubocop-powerhome
- Initial tracker implementation
- Extract tracker to a differnet file
- Extract DSL modules from tracker
- Add data tracking capabilities to trackers
- Do not install trackers for unexisting relations
- Move data tracker setup to dummy initializer
- Switch Rakefile to rubocop
- Remove combustion test database from git
- Add license_finder
- Add github workflow for data_tracker

### CODE REVIEW

**Comment by wadewinningham** [comment]
I feel like "Data Tracker" is a bit confusing as it's losing the connection to the Creator/Updater part. You consider other names for this? I think "Audit Tracker" reads a little better as the name is a bit more indicative of what it's really doing. I'm open to other names...

File: packages/data_tracker/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2022 Carlos Palhares
```
Comment (benlangfeld) [inline packages/data_tracker/LICENSE.txt:3]: Power

File: packages/data_tracker/spec/internal/app/models/internal/sale.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module Internal
+  class Sale < ::Internal::ApplicationRecord
+    track_data
```
Comment (benlangfeld) [inline packages/data_tracker/spec/internal/app/models/internal/sale.rb:5]: This is so generic it doesn't really mean anything. Can we use a name that has more meaning?

Comment (xjunior) [reply packages/data_tracker/spec/internal/app/models/internal/sale.rb:5]: @benlangfeld indeed, I didn't like this as well. Suggestions?

Comment (xjunior) [reply packages/data_tracker/spec/internal/app/models/internal/sale.rb]: How do you feel about @wadewinningham's suggestion: `AuditTracker` for the gem, and this helper method could be renamed to `track_audit_data`?

Comment (xjunior) [reply packages/data_tracker/spec/internal/app/models/internal/sale.rb]: /cc @benlangfeld

**Review by wadewinningham — DISMISSED** [review]

**Review by wadewinningham — APPROVED** [review]

**Pull request merged**